Source/Scene/Model.js

```diff
@@ -1896,13 +1949,20 @@ define([
         }
 
         if (model.extensionsUsed.WEB3D_quantized_attributes) {
-            vs = modifyShaderForQuantizedAttributes(vs, id, model);
+            var quantizedVs = quantizedVertexShaders[id];
+            if (!defined(quantizedVs)) {
+                quantizedVs = modifyShaderForQuantizedAttributes(vs, id, model);
+                quantizedVertexShaders[id] = quantizedVs;
+            }
+            vs = quantizedVs;
         }
 
-        var premultipliedAlpha = hasPremultipliedAlpha(model);
-        var finalFS = modifyShaderForColor(fs, premultipliedAlpha);
-        if (ClippingPlaneCollection.isSupported()) {
-            finalFS = modifyShaderForClippingPlanes(finalFS);
+        var finalFS = fs;
+        if (needsColorShadingCode(model)) {
+            finalFS = Model._modifyShaderForColor(finalFS, model._hasPremultipliedAlpha);
```

Comment:
The spec is currently spying on `Model._modifyShaderForColor`, does it work differently?
We're also currently caching the result from `hasPremultipliedApha` because the gltf asset usually gets released.